### PR TITLE
stubs and mocks

### LINKED ISSUE #4: Stubs
**What**

Develop the following method:

* `Rhum.Stub`

Method should be public and should return a stubbed method (like the one passed in) that should be able to tell the user if it was called

**Why**
This is a testing framework dammit!

**Checklist**

* Update README where needed

* Unit tests

* Maybe integration tests? Feel unit tests cover it fully

**Resources**

Use the following [issue](https://github.com/denoland/deno/issues/4092) as a guideline if possible

**Pseudo-code for Implementation**

```typescript
Stub: function (fn: Function): Function {
}

// Maybe used like so:
// Maybe something along the lines of:
let functionWasCalled = false;
function testFn() {
  functionWasCalled = true;
  return "Hello world!";
}
const copyTestFn = testFn;
Rhum.TestPlan("Testing TestCase", copyTestFn);
asserts.assertEquals(functionWasCalled, true);
```

### PR DESCRIPTION
**Pull request by crookse**
Closes #4 

For those outside of the Drash Land team, if you're reading this PR, the team and I agreed on the following definitions of stubs and mocks:

table

### CODE REVIEW

File: src/mock_builder.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        mock[method] = original[method];
+      }
+    });
+
+    return mock;
+  }
+
+  /**
+   * @description
+   *     Before constructing the mock object, track any constructur function args
```
Comment (ebebbington) [inline src/mock_builder.ts:77]: `constructur` --> `constructor`, only a minor typo

File: README.md [hunk 2]
```diff
@@ -248,6 +250,32 @@ Rhum.testPlan("My Plan", () => {
...
+
+Rhum.testPlan("My Plan", () => {
+  Rhum.testSuite("My Suite", () => {
+    Rhum.testCase("My Test Case", () => {
+      const mock = Rhum
+        .mock(Server)
+        .withConstructorArgs("arg1", "arg2", "arg3") // this call optional
+        .create();
+      Rhum.asserts.assertEquals(mock.protected_property, "a protected property");
+      Rhum.asserts.assertEquals(mock.protectedMethod(), "a protected method"); // does not run the console.log()
```
[the remark this answers is not in the document]
Comment (crookse) [reply README.md]: removed that comment (copypasta)

File: src/mock_builder.ts (hunk 1 above)
Comment (crookse) [reply src/mock_builder.ts:77]: :P

File: README.md [hunk 3]
```diff
@@ -248,6 +250,32 @@ Rhum.testPlan("My Plan", () => {
...
+}
+
+Rhum.testPlan("My Plan", () => {
+  Rhum.testSuite("My Suite", () => {
+    Rhum.testCase("My Test Case", () => {
+      const mock = Rhum
+        .mock(Server)
+        .withConstructorArgs("arg1", "arg2", "arg3") // this call optional
+        .create();
+      Rhum.asserts.assertEquals(mock.protected_property, "a protected property");
```
Comment (saragee3) [inline README.md:272]: curious why not `assets.equals` instead ?

**Pull request merged**